packages/cli/src/commands/alter.ts

```diff
@@ -227,19 +228,27 @@ export async function alter(
       }
 
       break;
-    case 'set-contract-address':
-      // find the steps that deploy contract
-      for (const actionStep in deployInfo.state) {
-        if (
-          deployInfo.state[actionStep].artifacts.contracts &&
-          deployInfo.state[actionStep].artifacts.contracts![targets[0]]
-        ) {
-          deployInfo.state[actionStep].artifacts.contracts![targets[0]].address = targets[1] as viem.Address;
-          deployInfo.state[actionStep].artifacts.contracts![targets[0]].deployTxnHash = '';
-        }
+    case 'set-contract-address': {
+      const [targetStepName, targetAddress] = targets;
+
+      if (!viem.isAddress(targetAddress)) {
+        throw new Error(`Invalid address given: "${targetAddress}"`);
+      }
+
+      // find the step that deploy contract
+      const [targetStep] = Object.values(deployInfo.state).filter((step) => !!step.artifacts?.contracts?.[targetStepName]);
+
+      if (!targetStep) {
+        throw new Error(`Could not find contract by step name "${targetStepName}"`);
       }
 
+      debug(`setting "${targetStepName}" address to "${targetAddress}"`);
+
+      targetStep.artifacts!.contracts![targetStepName].address = targetAddress;
```

Comment:
looks like the var shouldn't be called `targetStepName`, and instead called `targetContractName`